### PR TITLE
[JIT] Add Frozen Conv-> Add/Sub/Mul/Div fusion

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #50241 Re-enable no-op peepholes
* #50222 Add Post Freezing Optimizations, turn on by default in torch.jit.freeze
* #50221 Peephole Optimize out conv(x).dim(), which prevents BN fusion
* #50220 [JIT] Factor out peephole to own test file
* **#50075 [JIT] Add Frozen Conv-> Add/Sub/Mul/Div fusion**
* #50074 [JIT] Frozen Graph Conv-BN fusion

Adds Conv - Add/Sub/Mul/Div fusion for frozen models. This helps cover models like torchvision maskrcnn, which use a hand-rolled batchnorm implementation: https://github.com/pytorch/vision/blob/90645ccd0e774ad76200245e32222a23d09f2312/torchvision/ops/misc.py#L45. 

I haven't tested results yet but I would expect a somewhat similar speed up as conv-bn fusion (maybe a little less).

Differential Revision: [D25856265](https://our.internmc.facebook.com/intern/diff/D25856265)

### CODE REVIEW

File: torch/csrc/jit/passes/frozen_conv_folding.cpp [hunk 1]
```diff
@@ -107,10 +109,268 @@ void FoldFrozenConvBatchnorm(Block* b) {
...
+        auto mul_tensor = resizeConstantScalarOrTensorToShape(
+            mul_or_div->inputs().at(1), {out_channels}, bias.options());
+
+        mul_or_div->replaceInput(0, b->owningGraph()->insertConstant(bias));
+        mul_or_div->replaceInput(
+            1, b->owningGraph()->insertConstant(mul_tensor));
+
+        auto stack_out = runNodeIfInputsAreConstant(mul_or_div);
+        TORCH_INTERNAL_ASSERT(stack_out && stack_out->size() == 1);
+        Tensor fuse_bias = (*stack_out)[0].toTensor();
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/frozen_conv_folding.cpp:345]: Nit: in some places it's `at::Tensor`, here it's just `Tensor`.

File: torch/csrc/jit/passes/frozen_conv_folding.cpp [hunk 2]
```diff
@@ -107,10 +109,268 @@ void FoldFrozenConvBatchnorm(Block* b) {
...
+      "aten::sub.Scalar(Tensor self, Scalar other, Scalar alpha=1) -> Tensor",
+  };
+  return n->isMemberOf(add_set);
+}
+
+// In order to fuse add/sub/mul/div with conv, the dimensions of its
+// constant tensor must satisfy the following:
+// - with resizing, broadcast to w/ weight/bias tensor shape
+// - broadcast to the conv output shape
+// It needs to be have a shape that can resize to weight/bias
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/frozen_conv_folding.cpp:127]: Typo: "to be have".

File: torch/csrc/jit/passes/frozen_conv_folding.cpp [hunk 3]
```diff
@@ -107,10 +109,268 @@ void FoldFrozenConvBatchnorm(Block* b) {
...
+// tensor shape because we need to run the op with the conv
+// weights/bias without changing their sizes.
+// It needs to broadcast to the conv output shape so that we do
+// accidentally change the shape of op output by pre-fusing it
+// compared to eager.
+// The only dimension value shared by weight/bias/conv output
+// is they all contain a dim with value = channels-out. In the
+// conv output tensor, this is in the second dimension,
+// so the pointwise op tensor may have a second dimension of
+// value == chanels-out, but all the other dimensions have to be 1
```
Comment (ZolotukhinM) [inline torch/csrc/jit/passes/frozen_conv_folding.cpp:137]: Typo: chanels.

**Review by ZolotukhinM — APPROVED** [review]
Looks good, but I didn't dig deeply into the mathematical logic of the transformation.
